symplyphysics/core/vectors/arithmetics.py

```diff
@@ -194,3 +194,49 @@ def reject_cartesian_vector(
         original_vector_,
         scale_vector(-1, project_vector(original_vector_, target_vector_))
     )
+
+
+def diff_cartesian_vector_components(
```

Comment:
The thing is that the base vectors might also be dependent on time or some other parameter. In general, $`\dot{\mathbf{r}} = \dot{x} \mathbf{e}_x + \dot{y} \mathbf{e}_y + \dot{z} \mathbf{e}_z + x \dot{\mathbf{e}}_x + y \dot{\mathbf{e}}_y + z \dot{\mathbf{e}}_z`$ which is the case for rotating coordinate systems. And here I only want to have the coordinates differentiated.

Reply:
I think in our code we don't even have base vectors. They are simply indexes of array in coordinate system.

Reply:
`Vector([1, 0, 0])` is the base vector $`\mathbf{e}_x`$, why?

Reply:
Yes. What's wrong with it?

Reply:
You said in the beginning that it's vector derivative. But it's only true in a non-rotating coordinate system. And coordinate systems in general might perform arbitrary motion.

Reply:
We have a function to rotate coordinate system and vectors can be transformed to new coordinate system. You can take vector derivative is one coordinate system and transform it to rotated. There is no option to simultaneously rotate coordinate system and take vector derivative.
This topic is quite shady for me. Do you have any link to read about vector derivatives in rotating coordinate systems?

Reply:
Well you can read [this excerpt](https://www.southampton.ac.uk/~stefano/courses/PHYS2006/chapter4.pdf), for instance, right in the beginning (Section 4.1).

We seem to be talking about two different kinds of rotations of the coordinate system. What you're describing is the literal transition between two coordinate systems, one of which is found by rotating the other one by a certain angle. The coordinate systems might be static. Whereas what I'm talking about is that the coordinate system in question might be "fixed" to an rotating object and so it will be "rotating" about another, fixed coordinate system. The basic example of this is the Earth and the "local" coordinate system relative to its surface.

Reply:
I see. We do not support such `dynamic` coordinate systems. And do not intend to support them any time soon.

Reply:
Okay, I'll just add a note that this only works in non-rotating reference frames.